sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClient.cs

```diff
@@ -2159,6 +2160,30 @@ async Task<Page<T>> NextPageFunc(string nextLink, int? pageSizeHint)
             }
         }
 
+        /// <summary>
+        /// Queries for digital twins by iterating through a collection asynchronously.
+        /// </summary>
+        /// <param name="adtQueryBuilder"> Query object made with the ADT query builder. </param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The pageable list <see cref="AsyncPageable{T}"/> of query results.</returns>
+        /// <typeparam name="T">The type to deserialize the result to.</typeparam>
+        public virtual AsyncPageable<T> QueryAsync<T>(AdtQueryBuilder adtQueryBuilder, CancellationToken cancellationToken = default)
+        {
+            return QueryAsync<T>(adtQueryBuilder.GetQueryText(), cancellationToken);
```

Comment:
Should we consider a try/catch and log out to diagnostics the query text if there is an exception?